doc: update document for nuclei studio 2024.12

laomaolaile opened the pull request:
新增了Nuclei Studio 2024.12版相关文档

- 新增2024.12版更新说明
- 新增Flash Programming功能相关文档
- 新增了Nuclei NICE Wizard相关文档
- 新增Nuclei Model功能的使用相关文档
- 优化Nuclei Near Cycle Model相关文档
- 新增Live Watch功能相关文档

更新部分图片

Commit:
添加nice文档信息

Commit:
添加npk check说明

Commit:
添加Nuclei Studio 2024.12文档

fanghuaqi commented on source/ide/npkmanage.rst:
```diff
@@ -119,8 +119,52 @@ Type选择ssp: ``Soc Support Package``
+
+创建完成的软件包，在导入之前，可以先使用检查工具检测npk组件包的合法性，避免导入失败。其核心功能聚焦于代码风格、变量命名、函数结构以及依赖关系的合规性检查，确保代码质量。
+
+关于该工具的内容请参见: `软件包检查工具  <https://gito.corp.nucleisys.com/software/ide/npk-checker>`__ 
```
这个工具你怎么给的是内网的repo，要外网的 https://github.com/Nuclei-Software/npk-checker

其他的如果有类似内网的链接也请检查梳理 @laomaolaile

fanghuaqi commented on source/ide/npkmanage.rst:
```diff
@@ -119,8 +119,52 @@ Type选择ssp: ``Soc Support Package``
+.. |image30| image:: /asserts/nucleistudio/developer/image169.png
+
+
+2、当日志中出现类似There are file parsing exceptions提示时，可以去对应的`用户目录\nuclei-pack-npk-v2\CFGs\logs\sys\`路径下看当天的日志文件，进一步了解相关npk文件的问题点，以便进行相应的修正。
```
`用户目录\nuclei-pack-npk-v2\CFGs\logs\sys\` -> ``用户目录\nuclei-pack-npk-v2\CFGs\logs\sys\``

RST里面引用是 ``

fanghuaqi commented on source/ide/projectbuild.rst:
```diff
@@ -158,49 +158,95 @@ Nuclei Studio中编译Hello World项目
+**Load Program Image**
 
-.. |image4| image:: /asserts/nucleistudio/projectbuild/image5.png
+Load的文件，默认的elf格试的文件，也可以支持 ``*.bin、*.hex、*.s19、*.srec、*.symbolsrec`` 等各种格式
```
symbolsrec 这个是啥格式？

laomaolaile replied:
![image](https://github.com/user-attachments/assets/64faa82e-f9ab-4adf-a7c3-9fbf6035fba2)

这个选项编出来的文件就是*.symbolsrec的

fanghuaqi commented on source/ide/projectbuild.rst:
```diff
@@ -158,49 +158,95 @@ Nuclei Studio中编译Hello World项目
+|image20|
 
-.. |image7| image:: /asserts/nucleistudio/projectbuild/image8.png
+Reset and Run：选中时，Download命令会带上 ``rest`` 参数，这条指令在执行完load后可能强制系统复位（SRST），并让目标设备运行。
```
rest 是笔误？

fanghuaqi commented on source/ide/advanceusage.rst:
```diff
@@ -661,14 +661,149 @@ Nuclei Studio会在本地启一个web服务，同时打开Perfetto Trace Viewer
+
+新增指令  
+~~~~~~~~
+点击 ``Add...`` ,根据需要修改指令内容后，点击右上角 ``save`` 即可。
```
加一个空行

fanghuaqi commented on source/ide/advanceusage.rst:
```diff
@@ -661,14 +661,149 @@ Nuclei Studio会在本地启一个web服务，同时打开Perfetto Trace Viewer
+文件生成 
+~~~~~~~~
+
+可定义insn.h（包含内嵌汇编头文件）和 nice.cc（包含指令实现逻辑）文件的保存地址，点击Save and Generate File，会生成对应文件。
```
文件名引用包起来。

fanghuaqi commented on source/ide/advanceusage.rst:
```diff
@@ -661,14 +661,149 @@ Nuclei Studio会在本地启一个web服务，同时打开Perfetto Trace Viewer
+
+|image-nice-1|
+
+单个指令模板如上图所示，
```
这里需要看看排版出来的效果。改加粗和引用的都需要处理下。

fanghuaqi commented on source/ide/advanceusage.rst:
```diff
@@ -661,14 +661,149 @@ Nuclei Studio会在本地启一个web服务，同时打开Perfetto Trace Viewer
 在Nuclei Studio 2024.06版中，集成了Nuclei Near Cycle Model，它是由芯来科技自主研发的仿真测试和性能分析工具，可以帮助研发人员在项目初期进行一些必要的仿真测试和程序性能分析。
 
-Nuclei Near Cycle Model当前只有Linux版本，其具体介绍和命令行上使用参见 （https://doc.nucleisys.com/nuclei_tools/xlmodel/intro.html ） ，下面将在Nuclei Studio上演示如何使用Nuclei Near Cycle Model进行仿真和性能分析。
+Nuclei Near Cycle Modeld在Nuclei Studio 2024.06版中只有Linux版本，从2024.12版开始，已实现对Windows的支持。其具体介绍和命令行上使用参见 （https://doc.nucleisys.com/nuclei_tools/xlmodel/intro.html ） ，下面将在Nuclei Studio上演示如何使用Nuclei Near Cycle Model进行仿真和性能分析。
```
这里的链接可以直接引用到 sphinx内部的tag，而不需要给网页

fanghuaqi commented on source/ide/advanceusage.rst:
```diff
@@ -679,15 +814,49 @@ Nuclei Near Cycle Model对芯来全类型的Core都有支持，可以创建
 |image74|
 
-其中在Config options中需要配置 ``--trace=1 --gprof=1 --logdir=Debug`` , ``--trace=1`` 表示开启rvtrace， ``--gprof=1`` 表示开启gprof功能， ``--logdir=Debug`` 则表示最终生成的 ``.rvtrace`` 文件、 ``.gmon`` 文件存存放的路径为当前工程下的Debug目录。
+关于Model的参数如下图
```
这里的参数，不需要你来写一遍，你直接指向最新的文档里面的sphinx内部连接

fanghuaqi commented on source/ide/advanceusage.rst:
```diff
@@ -679,15 +814,49 @@ Nuclei Near Cycle Model对芯来全类型的Core都有支持，可以创建
+| ``--logdir=<str>`` | the directory to save trace and gprof files                                                     |
++--------------------+-------------------------------------------------------------------------------------------------+
+
+在演示示例的Config options中配置了 ``--trace=1 --gprof=1 --logdir=Debug --cpu=n300fd`` , ``--trace=1`` 表示开启rvtrace， ``--gprof=1`` 表示开启gprof功能， ``--logdir=Debug`` 则表示最终生成的 ``.rvtrace`` 文件、 ``.gmon`` 文件存存放的路径为当前工程下的Debug目录, ``--cup=n300fd`` 表示当前模拟的cpu核是n300fd。
```
这里保留，提示是按照上面的参数说明来进行的配置

fanghuaqi commented on source/ide/advanceusage.rst:
```diff
@@ -722,6 +887,163 @@ gprof工具在查看 ``.gmon`` 文件的同时，会根据其内容，解析出
+
+.. note::
+   
+   Live Watch功能依赖OpenOCD和telnet，仅支持使OpenOCD调试程序并开启了telnet端口的情况下使用Live Watch功能。
```
依赖Nuclei OpenOCD >= 2024.12版本。仅支持Nuclei CPU配置了RISC-V SBA功能

fanghuaqi commented on source/ide/advanceusage.rst:
```diff
@@ -722,6 +887,163 @@ gprof工具在查看 ``.gmon`` 文件的同时，会根据其内容，解析出
+
+   - 删除 Live Watch 视图中指定的变量行。
+
+**ReMove All**
```
这里怎么会是大写的ReMove , 应该是 Remove

fanghuaqi commented on source/ide/advanceusage.rst:
```diff
@@ -722,6 +887,163 @@ gprof工具在查看 ``.gmon`` 文件的同时，会根据其内容，解析出
+
+        - Save Data Speed : 设定 Live Watch 数据自动保存的频率，默认为每10分钟保存一次。
+
+**Number Fromat**
```
Fromat -> Format, 其他诸如类似的全局查找下

fanghuaqi commented on source/ide/intro.rst:
```diff
@@ -87,6 +87,72 @@ Nuclei Studio IDE充分利用上述Eclipse IDE优势，结合社区成熟的Ecli
+
+在Nuclei Studio 2024.12基于Eclipse Cpp 2024-12版本开发此版本。基础的CDT版本，升级到了11.6.1。
+
+升级RISC-V Toolchain、OpenOCD、QEMU版本
```
Nuclei Near Cycle Model版本也升级了，但是这个没有Release Note的链接。就主要是全面支持 Nuclei CPU 200，300,600,900,1000系列的CPU，同时支持Windows和Linux版本。

fanghuaqi reviewed (changes requested):
请按照上面的要求修改。同步也把其他的文件里面可以内部引用链接的地方清理下，包括qemu/gcc/model/openocd里面

Commit:
modify ide doc

fanghuaqi commented on source/ide/advanceusage.rst:
```diff
@@ -822,42 +823,10 @@ Nuclei Near Cycle Model采用Nuclei Studio中的Model运行配置来进行运
-
 在演示示例的Config options中配置了 ``--trace=1 --gprof=1 --logdir=Debug --cpu=n300fd`` , ``--trace=1`` 表示开启rvtrace， ``--gprof=1`` 表示开启gprof功能， ``--logdir=Debug`` 则表示最终生成的 ``.rvtrace`` 文件、 ``.gmon`` 文件存存放的路径为当前工程下的Debug目录, ``--cup=n300fd`` 表示当前模拟的cpu核是n300fd。
 
+关于Nuclei Near Cycle Model的参数具体说明，请参见 `Description of Parameters <https://doc.nucleisys.com/nuclei_tools/xlmodel/intro.html#description-of-parameters>`__  
```
这个链接在内部，他有相对路径。用sphinx的路径。

fanghuaqi commented:
@laomaolaile 文档需要继续更新

Commit:
Modify the link to the near syscel model

laomaolaile committed:
ci: upgrade github action version

Error: This request has been automatically failed because it uses a deprecated version of `actions/upload-artifact: v3`. Learn more: https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/

Signed-off-by: 张泽刚 <zhangzegang@whss1.corp.nucleisys.com>
